src/Raven.Server/Documents/Indexes/AbstractIndexCreateProcessor.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Client;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Exceptions.Documents.Compilation;
+using Raven.Client.Exceptions.Documents.Indexes;
+using Raven.Client.Util;
+using Raven.Server.Config;
+using Raven.Server.Documents.Indexes.MapReduce.OutputToCollection;
+using Raven.Server.Documents.Indexes.MapReduce.Static;
+using Raven.Server.Documents.Indexes.Persistence.Lucene;
+using Raven.Server.Documents.Indexes.Static;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Commands.Indexes;
+
+namespace Raven.Server.Documents.Indexes;
+
+public abstract class AbstractIndexCreateProcessor
```

Comment:
My proposals for different name:

- `AbstractIndexCreation` / `DatabaseIndexCreation` / `ShardIndexCreation`
- `AbstractIndexCreateDelegate` / `DatabaseIndexCreateDelegate` / `ShardIndexCreateDelegate`
- `AbstractIndexCreationMethods` / `DatabaseIndexCreationMethods` / `ShardIndexCreationMethods`